types/wlr_pointer_gestures_v1.c

```diff
@@ -0,0 +1,288 @@
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L
+#endif
+
+#include <assert.h>
+#include <stdlib.h>
+#include <wayland-server.h>
+#include <wlr/types/wlr_pointer.h>
+#include <wlr/types/wlr_pointer_gestures_v1.h>
+#include <wlr/util/log.h>
+#include "util/signal.h"
+#include "pointer-gestures-unstable-v1-protocol.h"
+
+#define POINTER_GESTURES_VERSION 1
+
+static void resource_handle_destroy(struct wl_client *client,
+		struct wl_resource *resource) {
+	wl_resource_destroy(resource);
+}
+
+static void resource_remove_from_list(struct wl_resource *resource) {
+	wl_list_remove(wl_resource_get_link(resource));
+}
+
+static const struct zwp_pointer_gestures_v1_interface gestures_impl;
+static const struct zwp_pointer_gesture_swipe_v1_interface swipe_impl;
+static const struct zwp_pointer_gesture_pinch_v1_interface pinch_impl;
+
+static struct wlr_pointer_gestures_v1 *pointer_gestures_from_resource(
+		struct wl_resource *resource) {
+	assert(wl_resource_instance_of(resource,
+				&zwp_pointer_gestures_v1_interface, &gestures_impl));
+	return wl_resource_get_user_data(resource);
+}
+
+static struct wlr_seat *wlr_seat_from_pointer_resource(
```

Comment:
We try to use the `wlr_` prefix only for exported functions. So it would be better to rename this `seat_from_pointer_resource`.